core/trino-main/src/main/java/io/trino/operator/project/InputChannels.java

```diff
@@ -51,6 +52,11 @@ public Page getInputChannels(Page page)
         return page.getColumns(inputChannels);
     }
 
+    public Page getLoadedInputChannels(Page page, Predicate<Integer> loadChannel)
```

Comment:
It would be better if `InputChannels` automatically loaded `Blocks/Pages` in `getInputChannels`